rs/boundary_node/ic_boundary/src/metrics.rs

```diff
@@ -651,18 +652,33 @@ pub async fn metrics_middleware(
             .observe(response_size as f64);
 
         // Anonymization
-        let s = anonymization_salt.load();
+        let salt = anonymization_salt.load();
 
-        let hash_fn = |v: &str| -> String {
-            if s.is_none() {
-                return "N/A".to_string();
-            }
+        let hash_fn = |input: &str| -> String {
+            let err_str = "N/A".to_string();
+
+            let salt = match salt.as_ref() {
+                Some(s) if s.len() == SALT_SIZE => s,
+                _ => return err_str,
+            };
+
+            let mut hasher = {
+                // Convert first 8 bytes to u64, return N/A if it fails
+                let key0 = match salt[0..8].try_into() {
+                    Ok(bytes) => u64::from_le_bytes(bytes),
+                    Err(_) => return err_str,
+                };
+                // Convert next 8 bytes to u64, return N/A if it fails
+                let key1 = match salt[8..16].try_into() {
+                    Ok(bytes) => u64::from_le_bytes(bytes),
+                    Err(_) => return err_str,
+                };
+                SipHasher::new_with_keys(key0, key1)
```

Comment:
It may be more convenient to use [new_with_key](https://docs.rs/siphasher/latest/siphasher/sip/struct.SipHasher.html#method.new_with_key) and avoid (or rather, delegate to the library) the `u64` conversions.

Reply:
yes, makes thanks